bitcoin/src/blockdata/script/borrowed.rs

```diff
@@ -218,6 +218,59 @@ impl Script {
         }
     }
 
+    /// Checks whether a script pubkey is a bare multisig output.
+    ///
+    /// In a bare multisig pubkey script the keys are not hashed, the script
+    /// is of the form:
+    ///
+    ///    `2 <pubkey1> <pubkey2> <pubkey3> 3 OP_CHECKMULTISIG`
+    #[inline]
+    pub fn is_multisig(&self) -> bool {
+        let required_sigs;
+
+        let mut instructions = self.instructions();
+        if let Some(Ok(Instruction::Op(op))) = instructions.next() {
+            if let Some(pushnum) = op.decode_pushnum() {
+                required_sigs = pushnum;
+            } else {
+                return false;
+            }
+        } else {
+            return false;
+        }
+
+        let mut num_pubkeys: u8 = 0;
+        while let Some(Ok(instruction)) = instructions.next() {
+            match instruction {
+                Instruction::PushBytes(_) => {
+                    num_pubkeys += 1;
+                }
+                Instruction::Op(op) => {
+                    if let Some(pushnum) = op.decode_pushnum() {
+                        if pushnum != num_pubkeys {
+                            return false;
+                        }
+                    }
+                    break;
+                }
+            }
+        }
```

Comment:
Since technically, OP_NEGATIVE1 and OP_1 through OP_16 are also considered "push operations"... I think this is technically incorrect if we don't care about pubkey length.

OP_3 pushes a single byte 0x03 on the stack, and OP_CHECKMULTISIG would consume it.

Since the N and M values must be between OP_1 and OP_16, they are able to be narrowed down, but OP_NEGATIVE1 would be a "pubkey" of `0x81`

So:

1. First instruction MUST be OP 1-16
2. Take instructions both PushBytes and Op `-1, 1-16` as pubkey counts until you hit OP_CHECKMULTISIG
3. Parse the last instruction before MULTISIG as the number of pubkeys.

Might want to use `self.instructions().peekable()`

Reply:
In practice tho, including these obviously incorrect edge cases (for now... as long as a 0x08 one byte pubkey algorithm doesn't get added in a hardfork or something lol) is not a necessity IMO.

If one person agrees that this is ok to ignore, I'll also give my ACK.

Reply:
For what its worth, I don't think we should add complexity to the code to make it even more general to cover these scripts that will fail at execution.

Reply:
I also agree that we shouldn't bother trying to deal with this case.

Reply:
There are also cases where n > 20 is not a valid argument for OP_CHECKMULTISIG. Should that be considered?